### PR TITLE
Compile AP_Generator_Richenpower in by default

### PR DESCRIPTION
**Pull request by Tabocim**
Solving issue #22368 regarding RICHENPOWER GENERATOR's driver

### CODE REVIEW

**Comment by peterbarker** [comment]
@Tabocim of all our drivers, this is the only one which is off by default for everything - on the basis that very few people actually need it.

If you need to use the driver, you're probably the sort of people that need to use something along the lines of https://ardupilot.org/copter/docs/common-oem-customizations.html to create yourself a custom build setup.

If you do that, you can drop `define AP_GENERATOR_RICHENPOWER_ENABLED 1` in the bottom of your custom hwdef.

tldr; the pattern you're trying to correct here is actually by design.

**Comment by onurerden** [comment]
Hi @peterbarker ,
Actually me too, using RichenPower generators, and It took "a lot of time" to realize the driver was disabled with 4.3.0. Spent too much time on uart cable between generator and flight controller, and it was not stated in changelogs on version 4.3.0. Since it was working well with its features, I don't see any reason to disable it with an update, unless there was a critical feedback from users.
As you said "this is the only one which is off by default". There are lots of different equipments (lidars, positioning systems, radars etc.) which are not using by 90% of all users, but they are not disabled by build.
Again, I understand what oem customization is, but I don't understand why it was disabled while it was enabled with previous version.
My vote is for accepting the change. I kindly ask people who has authorization to consider this again. Thanks.

**Comment by peterbarker** [comment]
> Hi @peterbarker , Actually me too, using RichenPower generators, and It took "a lot of time" to realize the driver was disabled with 4.3.0. Spent too much time on uart cable between generator and flight controller, and it was not stated in changelogs on version 4.3.0. Since it was working well with its features, I don't see any reason to disable it with an update, unless there was a 

Should definitely have been pulled out in changelog notes - apologies for that.

> critical feedback from users. As you said "this is the only one which is off by default". There are lots of different equipments (lidars, positioning systems, radars etc.) which are not using by 90% of all users, but they are not disabled by build. Again, I understand what oem customization is, but I don't understand why it was disabled while it was enabled with previous version. My vote is for accepting the change. I kindly ask people who has authorization to consider this again. Thanks.

It's a matter of running short on flash, even on 2MB boards - nothing more than that.  I figured we should really be adopting a pattern where these niche drivers which will only be run on *serious* Copters aren't in by default so we still have room for more commonly-used features (e.g. all of the filtering stuff we've gotten over the last couple of years).

I'm surprised at how many people this has bitten.  So I've marked this as a DevCall topic for discussion.  At the very least we need to make the information you gleaned the hard way much easier to find!

**Comment by peterbarker** [comment]
I've fixed both the commit message and the title of this PR to give a bit more information as to what's going on.

**Comment by rmackay9** [comment]
We discussed this during today's dev call but decided that we cannot merge this PR as-is because it breaks the build for many of the smaller boards.  I think PeterB has created an alternative PR here https://github.com/ArduPilot/ardupilot/pull/22389 so I will close this one if that's OK. Txs very much for highlighting this problem and contributing a fix.

**Comment by peterbarker** [comment]
OK, so we brought this up on DevCall and it was generally agreed that this should be a custom-build-server / custom-build / OEM-build-setup thing rather than being enabled by default.

I've created https://github.com/ArduPilot/ardupilot/pull/22389 to enable it to be included in builds made from our custom build server.

@Hwurzburg will be adding a note to the Wiki to raise visibility of the fact it's not enabled by default.

If you can think of other ways we can improve things - without costing several kB of flash for the 99.999% of our users who don't use RichenPower generators - please do mention them.
